statsmodels/base/optimizer.py

```diff
@@ -156,11 +155,14 @@ def _fit(self, objective, gradient, start_params, fargs, kwargs,
         extra_fit_funcs = kwargs.setdefault('extra_fit_funcs', dict())
 
         methods = ['newton', 'nm', 'bfgs', 'lbfgs', 'powell', 'cg', 'ncg',
-                'basinhopping']
+                'basinhopping', 'dogleg']
```

Comment:
I thought of adding here a `'minimize'` keyword directly, so we can use whatever scipy minimize provides without having to hardcode any methods. 
There user will have to provide the name of the actual method, either in the **keywords, or we could try parsing a double method here e.g. `'minimize-dogleg'.split('-')

adding some specific methods like you did with dogleg is also possible